hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -344,6 +340,9 @@ protected final List<Path> compact(final CompactionRequestImpl request,
     boolean finished = false;
     List<StoreFileScanner> scanners =
       createFileScanners(request.getFiles(), smallestReadPoint, dropCache);
+    T writer = null;
+    CompactionProgress progress = new CompactionProgress(fd.maxKeyCount);
+    progressMap.put(request, progress);
```

Comment:
Could warn if the return value from `put(...)` was anything but `null` (i.e. we should not have a progress for this `CompactionRequestImpl`)

Reply:
Right, the null handling was unfinished. Same for your other point as well. But I think we will be going with @Apache9 's alternative.